Source/NSSortDescriptor.m

```diff
@@ -113,8 +113,16 @@ - (NSComparisonResult) compareObject: (id) object1 toObject: (id) object2
   id comparedKey1 = [object1 valueForKeyPath: _key];
   id comparedKey2 = [object2 valueForKeyPath: _key];
 
-  result = (NSComparisonResult) [comparedKey1 performSelector: _selector
-                                                   withObject: comparedKey2];
+  if(_comparator != NULL)
```

Comment:
The code would be easier to read if you used the same order in both checks. That is either (_comparator != NULL) or (_comparator == NULL) both here and in line 149. Also a space after the "if" is required.